manual/src/005_protocol-specification-rules.md

```diff
@@ -493,6 +493,43 @@ Thus, the complete example becomes:
 Note that the protocol description only specifies a model, but not which
 properties it might satisfy. We discuss these in the next section.
 
+### Clustering by Agent
+
+To enhance the clarity of the graph and identify the agents involved, nodes can be grouped by their respective agents. Clusters are purely a matter of display and do not affect the reasoning Tamarin uses to construct the proof. This is a purely aesthetic option, making the graph easier to read, which can be enabled or disabled by the user. The clusters will be displayed as colored boxes containing the relevant nodes. 
+
+The heuristic used for clustering identifies different sessions by checking if two nodes have the same agent and are connected by an edge; they will then be part of the same cluster. This method simplifies the organization of agents within the graph and ensures a clear representation of the agents and their respective nodes. The rules within a cluster will take the color of the cluster.
+
+There are two methods to group by agent:
+
+#### Manual Clustering
+
+You can manually specify the agents representing one or more rules. These can be any rules. The format for manual clustering is as follows:
```

Comment:
On can manually specify the rule(s) representing the behavior of an agent. For this, the rules are annotated with the agent's name as follows: